src/main/fc/rc.c

```diff
@@ -148,6 +148,24 @@ float applyRaceFlightRates(const int axis, float rcCommandf, const float rcComma
     return angleRate;
 }
 
+float applyKissRates(const int axis, float rcCommandf, const float rcCommandfAbs)
+{
+    float kissRpyUseRates;
+    const float rcCurvef = currentControlRateProfile->rcExpo[axis] / 100.0f;
+
+    if (currentControlRateProfile->rates[axis] < 100) {
+        kissRpyUseRates = 1 - rcCommandfAbs * currentControlRateProfile->rates[axis] / 100.0f;
+    } else {
+        //prevent division by zero
+        kissRpyUseRates = 1 - rcCommandfAbs * 0.99f;
+    }
+    
+    rcCommandf = (power3(rcCommandf) * rcCurvef + rcCommandf * (1 - rcCurvef)) * (currentControlRateProfile->rcRates[axis] / 1000.0f);
+    float kissAngle = constrainf(((2000.0f * (1.0f / kissRpyUseRates)) * rcCommandf), -2000.0f, 2000.0f);
```

Comment:
Rather than hardcoded `2000` use `SETPOINT_RATE_LIMIT`.

Reply:
ok

Reply:
but `SETPOINT_RATE_LIMIT` is 1998 and slightly changes the formula